crates/bevy_reflect/src/array.rs

```diff
@@ -0,0 +1,207 @@
+use crate::{serde::Serializable, Reflect, ReflectMut, ReflectRef};
+use std::{
+    any::Any,
+    hash::{Hash, Hasher},
+};
+
+/// A static-sized array of [`Reflect`] items.
+/// This corresponds to types like `[T; N]` (arrays)
+pub trait Array: Reflect {
+    fn get(&self, index: usize) -> Option<&dyn Reflect>;
+    fn get_mut(&mut self, index: usize) -> Option<&mut dyn Reflect>;
+    fn len(&self) -> usize;
+    fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+    fn iter(&self) -> ArrayIter;
+    fn clone_dynamic_array(&self) -> DynamicArray {
```

Comment:
I have a small preference for `Array::clone_dynamic(a)` and `List::clone_dynamic(b)` instead of `a.clone_dynamic_array()` and `b.clone_dynamic_list()`.

Reply:
I mostly just wanted to avoid the name clashing, however, I guess you can just fully qualify the function :)